org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/ui/tests/views/SmokeViewsTest.java

```diff
@@ -102,6 +104,36 @@ public void testOpenBytecodeReferenceView() throws Exception {
 		smokeTest("org.eclipse.jdt.bcoview.views.BytecodeReferenceView");
 	}
 
+	@Test
+	public void testOpenJavadocView() throws Exception {
+		smokeTest(JavaUI.ID_JAVADOC_VIEW);
+	}
+
+	@Test
+	public void testOpenPackagesView() throws Exception {
+		smokeTest(JavaUI.ID_PACKAGES_VIEW);
+	}
+
+	@Test
+	public void testOpenTypesView() throws Exception {
+		smokeTest(JavaUI.ID_TYPES_VIEW);
+	}
+
+	@Test
+	public void testOpenMembersView() throws Exception {
+		smokeTest(JavaUI.ID_MEMBERS_VIEW);
+	}
+
+	@Test
+	public void testOpenProjectsView() throws Exception {
+		smokeTest(JavaUI.ID_PROJECTS_VIEW);
+	}
+
+	@Test
+	public void testOpenSourceView() throws Exception {
+		smokeTest(JavaUI.ID_SOURCE_VIEW);
+	}
+
 	private void smokeTest(String viewId) throws PartInitException {
 		view = window.getActivePage().showView(viewId);
```

Comment:
Looking on the test now I wonder why I haven't added error log listener for platform in the setup and checked for new errors reported after the test. May be you could add that? In case view opens but reports some errors in the log.

Reply:
I would rather fix the more pressing issue now and leave that for some later point (and probably someone else). Incremental fixes and enhancements .